neo/rawio/spikeglxrawio.py

```diff
@@ -0,0 +1,315 @@
+        if device is None:
+            if len(self._channel_location) == 1:
+                locations = list(self._channel_location.values())[0]
+            else:
+                raise ValueError('device must specified')
+        else:
+            locations = self._channel_location[seg_index, device]
+        locations = locations * [[x_pitch, y_pitch]]
+        return locations
+
+
+def scan_files(dirname):
+    """
+    Scan pair of bin/meta files and return information about it.
+    """
+    info_list = []
+
+    for root, dirs, files in os.walk(dirname):
+
+        for file in files:
+            if not file.endswith('.meta'):
+                continue
+            meta_filename = Path(root) / file
+            bin_filename = Path(root) / file.replace('.meta', '.bin')
+            meta = read_meta_file(meta_filename)
+
+            num_chan = int(meta['nSavedChans'])
+
+            # when file is Noise4Sam_g0_t0.nidq.bin or Noise4Sam_g0_t0.imec0.lf.bin
+            # name is the first part "Noise4Sam_g0_t0"
+            # gtX X is the seg_index here 0
+            # nidq or imec0 is the device
+            # lf or ap is "signal_kind"
+            # stream_name = device + signal_kind
+            name = file.split('.')[0]
+            r = re.findall(r'_g(\d*)_t', name)
+            seg_index = int(r[0][0])
+            device = file.split('.')[1]
+            if 'imec' in device:
+                signal_kind = file.split('.')[2]
+                stream_name = device + '.' + signal_kind
+                units = 'uV'
+                # please note the 1e6 in gain for this uV
+
+                # metad['imroTbl'] contain two gain per channel  AP and LF
+                # except for the last fake channel
+                per_channel_gain = np.ones(num_chan, dtype='float64')
+                if signal_kind == 'ap':
+                    index_imroTbl = 3
+                elif signal_kind == 'lf':
+                    index_imroTbl = 4
+                for c in range(num_chan - 1):
+                    # the last channel don't have gain
+                    per_channel_gain[c] = 1. / float(meta['imroTbl'][c].split(' ')[index_imroTbl])
+                gain_factor = float(meta['imAiRangeMax']) / 512
+                channel_gains = per_channel_gain * gain_factor * 1e6
+
+            else:
+                signal_kind = ''
+                stream_name = device
+                units = 'V'
+                channel_gains = np.ones(num_chan)
+
+                # there differents kind of channel with diffrents gain
+                mn, ma, xa, dw = [int(e) for e in meta['snsMnMaXaDw'].split(sep=',')]
+                per_channel_gain = np.ones(num_chan, dtype='float64')
+                per_channel_gain[0:mn] = float(meta['niMNGain'])
+                per_channel_gain[mn:mn + ma] = float(meta['niMAGain'])
+                gain_factor = float(meta['niAiRangeMax']) / 32768
```

Comment:
Where does the `32768` come from? Is this a global neuropixel feature that is consistent across all systems. Would it be possible to extract this from some recording file instead of hardcoding here?

Reply:
This come from the doc and also come from the code proposed by in the spikeglx project.

Reply:
Maybe it would be good to put a comment here with the reference, so that if something strange is happening, we can quickly check if this number changed...

Reply:
Hi @samuelgarcia, so you are referring to this code [here](https://github.com/billkarsh/SpikeGLX)? The constant pops up in multiple contexts in this code, can you be more precise to which of those the constant here corresponds?